flink-cdc-common/src/test/java/org/apache/flink/cdc/common/factories/FactoryHelperTests.java

```diff
@@ -0,0 +1,180 @@
+import org.apache.flink.cdc.common.configuration.ConfigOptions;
+import org.apache.flink.cdc.common.configuration.Configuration;
+import org.apache.flink.table.api.ValidationException;
+
+import org.apache.flink.shaded.guava31.com.google.common.collect.Sets;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+/** Tests for {@link FactoryHelper}. */
+public class FactoryHelperTests {
+
+    private Factory getDummyFactory() {
+
+        return new Factory() {
+            @Override
+            public String identifier() {
+                return "dummy";
+            }
+
+            @Override
+            public Set<ConfigOption<?>> requiredOptions() {
+                return Sets.newHashSet(
+                        ConfigOptions.key("id").intType().noDefaultValue(),
+                        ConfigOptions.key("name").stringType().noDefaultValue(),
+                        ConfigOptions.key("age").doubleType().noDefaultValue());
+            }
+
+            @Override
+            public Set<ConfigOption<?>> optionalOptions() {
+                return Sets.newHashSet(
+                        ConfigOptions.key("hobby").stringType().noDefaultValue(),
+                        ConfigOptions.key("location").stringType().defaultValue("Everywhere"),
+                        ConfigOptions.key("misc")
+                                .mapType()
+                                .defaultValue(Collections.singletonMap("A", "Z")));
+            }
+        };
+    }
+
+    @Test
+    public void testCorrectConfigValidation() {
+        // This is a valid configuration.
+        Map<String, String> configurations = new HashMap<>();
+        configurations.put("id", "1");
+        configurations.put("name", "Alice");
+        configurations.put("age", "17");
+        configurations.put("location", "Here");
+
+        FactoryHelper factoryHelper =
+                FactoryHelper.createFactoryHelper(
+                        getDummyFactory(),
+                        new FactoryHelper.DefaultContext(
+                                Configuration.fromMap(configurations), null, null));
+
+        factoryHelper.validate();
+    }
+
+    @Test
+    public void testMissingRequiredOptionConfigValidation() {
+        // This configuration doesn't provide all required options.
+        Map<String, String> configurations = new HashMap<>();
+        configurations.put("id", "1");
+        configurations.put("age", "17");
+        configurations.put("location", "Here");
+
+        FactoryHelper factoryHelper =
+                FactoryHelper.createFactoryHelper(
+                        getDummyFactory(),
+                        new FactoryHelper.DefaultContext(
+                                Configuration.fromMap(configurations), null, null));
+
+        Assertions.assertThrowsExactly(
```

Comment:
I'd suggest to use the `assertj` assertions, but if you cherry pick from other PR should be fine if it is used there

Reply:
Thanks! I've pushed another commit to clean up assertions.